UniWatch/Controllers/ClassController.cs

```diff
@@ -9,75 +9,74 @@
         {
             _dataAccess = dataAccess;
         }
+
         //GET: Index
         /// <summary>
         /// Displays all classes taught by the teacher
         /// </summary>
-        /// <param name="teacherId">Id of the teacher</param>
-        public ActionResult Index(int teacherId)
+        [OverrideAuthorization]
+        [Authorize]
+        public ActionResult Index()
         {
-            var teacher = _dataAccess.UserManager.GetTeacherById(teacherId);
-                if (teacher==null)
+            // TODO: modify view to show appropriate information for either user type
+            var user = _dataAccess.UserManager.GetUser();
+            if(user is Teacher)
+            {
+                Teacher teacher = user as Teacher;
+                return View(_dataAccess.ClassManager.GetClassesForTeacher(teacher.Id));
+            }
+            else
             {
-                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
+                Student student = user as Student;
+                return View(_dataAccess.ClassManager.GetClassesForStudent(student.Id));
             }
-            ViewBag.TeacherId = teacherId;
-            return View(_dataAccess.ClassManager.GetClassesForTeacher(teacherId));
         }
 
-
         // GET: Create
         /// <summary>
-        /// Sends user to the create class page
+        /// Returns the view to create a new class
         /// </summary>
-        /// <param name="teacherId">Id of the teacher</param>
-        public ActionResult Create(int teacherId)
+        public ActionResult Create()
         {
-            var teacher = _dataAccess.UserManager.GetTeacherById(teacherId);
-            if (teacher == null)
-            {
-                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
-            }
-           // var teacher = _dataAccess.UserManager.GetTeacherById(teacherId);
-            return View(new Class() { Teacher = teacher });
+            return View();
         }
 
         //POST: Create
         /// <summary>
         /// Create class for the teacher
         /// </summary>
         /// <param name="class">Model containing information to create the class</param>
-        /// <param name="teacherId">Id of the teacher</param>
         [HttpPost]
         [ValidateAntiForgeryToken]
-        public ActionResult Create([Bind(Include = "Name,Number,Section,Semester,Year")] Class @class, int teacherId)
+        public ActionResult Create([Bind(Include = "Name,Number,Section,Semester,Year")] Class @class)
         {
-            if (!ModelState.IsValid)
+            if(!ModelState.IsValid)
             {
                 return View(@class);
             }
             try
             {
-                var created = _dataAccess.ClassManager.CreateClass(@class.Name, @class.Number, @class.Section, @class.Semester, @class.Year, teacherId);
+                var teacher = _dataAccess.UserManager.GetUser() as Teacher;
```

Comment:
Teacher can't be null. It requires that a user is logged in. The control wide authorization restricts it to teachers only.